numpyro/distributions/util.py

```diff
@@ -244,7 +246,10 @@ def _multinomial(key, p, n, n_max, shape=()):
 
 
 def multinomial(key, p, n, shape=()):
-    n_max = int(jnp.max(n))
+    assert not isinstance(
+        n, jax.core.Tracer
+    ), "The total count parameter `n` should not be a jax abstract array."
+    n_max = int(np.max(jax.device_get(n)))
```

Comment:
This covers more usage cases while raising an early error if n is an abstract array.